src/components/GameBoard.js

```diff
@@ -1,123 +1,128 @@
+    image: '🦧'
+  },
+  {
+    type: 'faceHalo',
+    image: '😇'
+  },
+  {
+    type: 'faceWink',
+    image: '😉'
+  },
+  {
+    type: 'animalGorilla',
+    image: '🦍'
+  },
+  {
+    type: 'animalPoodle',
+    image: '🐩'
+  }
+  ,
+  {
+    type: 'animalMouse',
+    image: '🐭'
+  }
 ];
 
-const createShuffledDeck = deck => {
-    deck = [...deck, ...deck];
-    let count = deck.length;
-    while (count > 0) {
-        let randomPick = Math.floor(Math.random() * count);
-        count--;
-        [deck[count], deck[randomPick]] = [deck[randomPick], deck[count]];
-    }
-    return deck.map(each => ({ ...each, 'id': uuidv4() }));
+const shuffleAndCreateDeck = deck => {
+  let count = deck.length;
+  while (count > 0) {
+    const randomPick = Math.floor(Math.random() * count);
+    count--;
+    [deck[count], deck[randomPick]] = [deck[randomPick], deck[count]];
+  }
+  return deck.map(eachCard => ({ ...eachCard, 'id': uuidv4() }));
 }
 
-export const GameBoard = () => {
-    const [inactiveCards, setInactiveCards] = useState([]);
-    const [disableAllClick, setDisableAllClick] = useState(false);
-    const [openCards, setOpenCards] = useState([]);
-    const [numberOfMoves, setNumberOfMoves] = useState(0);
-    const [cards, setCards] = useState(() =>
-        createShuffledDeck(cardPool)
-    );
-
-    useEffect(() => {
-        if (openCards.length === 2) {
-            setDisableAllClick(true);
-            setTimeout(() => { evaluate() }, 3500);
-            setDisableAllClick(true);
-        }
-    }, [openCards])
+const GameBoard = () => {
+  const [inactiveCards, setInactiveCards] = useState([]);
+  const [disabled, setDisabled] = useState(false);
+  const [openedCards, setOpenedCards] = useState([]);
+  const [numberOfMoves, setNumberOfMoves] = useState(0);
+  const [cards, setCards] = useState(() =>
+    shuffleAndCreateDeck([...cardPool, ...cardPool])
+  );
 
-    const restartGame = () => {
-        setCards(createShuffledDeck(cardPool));
-        setInactiveCards([]);
-        setOpenCards([]);
-        setNumberOfMoves(0);
+  useEffect(() => {
+    if (openedCards.length === 2) {
+      setDisabled(true);
+      setTimeout(() => { evaluate() }, 3500);
     }
+  }, [openedCards])
 
-    const handleCardClicked = (clickedCardId) => {
-        setNumberOfMoves(numberOfMoves => numberOfMoves + 1);
-        setOpenCards([...openCards, clickedCardId])
-    }
+  const restartGame = () => {
+    setCards(shuffleAndCreateDeck([...cardPool, ...cardPool]));
+    setInactiveCards([]);
+    setOpenedCards([]);
+    setNumberOfMoves(0);
+  }
 
-    const checkCardsType = (card1Id, card2Id, deckArray) => {
-        let card1Obj = deckArray.filter(card => card.id === card1Id);
-        let card2Obj = deckArray.filter(card => card.id === card2Id);
-        return (card1Obj[0].type === card2Obj[0].type) ? true : false;
-    }
+  const handleCardClicked = (cardId) => {
+    setNumberOfMoves(numberOfMoves => numberOfMoves + 1);
```

Comment:
You don't need to access previous object right now. A simple `setNumberOfMoves(numerOfMoves + 1)` will do the job.